internal/middlewares/delay.go

```diff
@@ -0,0 +1,25 @@
+package middlewares
+
+import (
+	"time"
+
+	"github.com/valyala/fasthttp"
+)
+
+func ArbitraryDelay(delay time.Duration) Basic {
+	return func(next fasthttp.RequestHandler) (handler fasthttp.RequestHandler) {
+		return func(ctx *fasthttp.RequestCtx) {
+			started := time.Now()
+
+			next(ctx)
+
+			delta := delay - time.Since(started)
+
+			if delta < 0 {
```

Comment:
Can we add a unit test for this scenario? I understand this is likely a scenario that will cannot occur.